File: scenarios/identity-python-default-azure-credential/golden/identity.py
from __future__ import annotations

import logging
import os
import sys

from azure.core.exceptions import ClientAuthenticationError, HttpResponseError
from azure.identity import DefaultAzureCredential
from azure.keyvault.secrets import SecretClient


def configure_identity_diagnostics() -> None:
    logging.basicConfig(
        level=logging.INFO,
        format="%(asctime)s %(levelname)s %(name)s: %(message)s",
    )
    logging.getLogger("azure.identity").setLevel(logging.INFO)


def run() -> int:
    vault_url = os.environ["AZURE_KEY_VAULT_URL"]
    secret_name = os.environ["AZURE_KEY_VAULT_SECRET_NAME"]
    credential = DefaultAzureCredential()

    try:
        with credential:
            with SecretClient(
                vault_url=vault_url,
                credential=credential,
            ) as client:
                secret = client.get_secret(secret_name)
                print(secret.value)
    except ClientAuthenticationError as error:
        print(f"Azure authentication failed: {error}", file=sys.stderr)
        return 1
    except HttpResponseError as error:
        print(f"Key Vault request failed: {error}", file=sys.stderr)
        return 1
    return 0


def main() -> int:
    configure_identity_diagnostics()
    try:
        return run()
    except KeyError as error:
        print(
            f"Missing required environment variable: {error.args[0]}",
            file=sys.stderr,
        )
        return 2


if __name__ == "__main__":
    raise SystemExit(main())

File: scenarios/identity-typescript-default-azure-credential/golden/src/identity.ts
import {
  AuthenticationError,
  DefaultAzureCredential,
} from "@azure/identity";
import { SecretClient } from "@azure/keyvault-secrets";
import { setLogLevel } from "@azure/logger";

declare const process: {
  env: Record<string, string | undefined>;
};

function requireEnvironment(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Set ${name} before running the application.`);
  }
  return value;
}

async function main(): Promise<void> {
  setLogLevel("info");

  const credential = new DefaultAzureCredential({
    managedIdentityClientId: process.env.AZURE_CLIENT_ID,
  });
  const client = new SecretClient(
    requireEnvironment("AZURE_KEY_VAULT_URL"),
    credential,
  );

  try {
    const secret = await client.getSecret(
      requireEnvironment("AZURE_KEY_VAULT_SECRET_NAME"),
    );
    console.log(secret.value);
  } catch (error: unknown) {
    if (error instanceof AuthenticationError) {
      console.error("Azure authentication failed.", error.message);
    }
    throw error;
  }
}

await main();
